src/config/configProcessors.js

```diff
@@ -0,0 +1,131 @@
+goog.module('measurementLibrary.config.configProcessors');
+const GoogleAnalyticsEventProcessor = goog.require('measurementLibrary.eventProcessor.GoogleAnalyticsEventProcessor');
+const {LogLevel, log} = goog.require('measurementLibrary.logging');
+const DataLayerHelper = goog.require('dataLayerHelper.helper.DataLayerHelper');
+
+/**
+ * @typedef {function(new:StorageInterface, !Object<string, *>)}
+ */
+let StorageConstructor;
+
+/**
+ * @typedef {function(new:EventProcessor, !Object<string, *>)}
+ */
+let ProcessorConstructor;
+
+/**
+ * A map assigning names to all of the built in storage interfaces.
+ * @const {!Object<string, !StorageConstructor>}
+ */
+const DEFAULT_STORAGE_INTERFACES = {};
+
+/**
+ * A map assigning names to all of the built in event processors.
+ * @const {!Object<string, !ProcessorConstructor>}
+ */
+const DEFAULT_EVENT_PROCESSORS =
+    {'googleAnalytics': GoogleAnalyticsEventProcessor};
+
+/**
+ * Given a string, return the constructor saved in
+ * DEFAULT_EVENT_PROCESSORS or DEFAULT_STORAGE_INTERFACES, depending on
+ * isEventProcessor. Constructor objects are passed through without
+ * modification.
+ *
+ * @param {!ProcessorConstructor|!StorageConstructor|string} obj
+ * @param {boolean} isEventProcessor True if obj represents an event processor,
+ *     false if obj represents storage
+ * @return {!ProcessorConstructor|!StorageConstructor}
+ */
+function getConstructor_(obj, isEventProcessor) {
```

Comment:
I think this is a little error prone, and hard to understand.

Instead, I'd split this into two functions that return a fully constructed object, rather than the constructor. At the call-site, you'd just do:

`const processor = buildProcessor_(processorCtorObj, processorCtorParams);`

What are your thoughts?

Reply:
Sure, that seems easier to understand, especially since it's not likely that the current functions would be called separately.